### PR TITLE
docs: invalid use of `Uri.https`

### LINKED ISSUE #3676: docs: invalid use of `Uri.https` in readme
The README has the following snippet
```dart
@riverpod
Future<String> boredSuggestion(BoredSuggestionRef ref) async {
  final response = await http.get(
    Uri.https('https://boredapi.com/api/activity'),
  );
  final json = jsonDecode(response.body);
  return json['activity']! as String;
}
```
which is invalid use of `Uri.https`... likely should just be `Uri.parse` instead
```diff
@riverpod
Future<String> boredSuggestion(BoredSuggestionRef ref) async {
  final response = await http.get(
-   Uri.https('https://boredapi.com/api/activity'),
+   Uri.parse('https://boredapi.com/api/activity'),
  );
  final json = jsonDecode(response.body);
  return json['activity']! as String;
}
```

~~and the Bored-API used in the docs (and README) is down https://github.com/drewthoennes/Bored-API/issues/83. I think it has been for about a month.~~

### PR DESCRIPTION
**Pull request by Luckey-Elijah**
## Related Issues

fixes #3676

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`).

- [x] I have updated the `CHANGELOG.md` of the relevant packages.
      Changelog files must be edited under the form:

  ```md
  ## Unreleased fix/major/minor

  - Description of your change. (thanks to @yourGithubId)
  ```

- [x] If this contains new features or behavior changes,
      I have updated the documentation to match those changes.

### CODE REVIEW

File: packages/riverpod/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,7 @@
+## 2.5.2
+
```
Comment (rrousselGit) [inline packages/riverpod/CHANGELOG.md:2]: Website changes don't involve changelogs :)

File: README.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ Long story short:
   @riverpod
   Future<String> boredSuggestion(BoredSuggestionRef ref) async {
     final response = await http.get(
-      Uri.https('https://boredapi.com/api/activity'),
+      Uri.parse('https://boredapi.com/api/activity'),
```
Comment (rrousselGit) [inline README.md:51]: What about this?

```suggestion
      Uri.https('boredapi.com', 'api/activity'),
```

**Comment by Luckey-Elijah** [comment]
not sure why [job/27921668359](https://github.com/rrousselGit/riverpod/actions/runs/10097255732/job/27921668359?pr=3677) fails if i only change docs

**Comment by rrousselGit** [comment]
custom_lint is broken atm.
I'm working on fixing it, but unfortunately it's a lot of work. Ignore it :)

**Comment by rrousselGit** [comment]
Thanks for this!

**Pull request merged**